funcx_endpoint/funcx_endpoint/executors/__init__.py

```diff
@@ -1,3 +1,5 @@
+from funcx_endpoint.executors.gc_executor.gc_executor import GCExecutor
 from funcx_endpoint.executors.high_throughput.executor import HighThroughputExecutor
+from funcx_endpoint.executors.process_pool.process_pool import ProcessPoolExecutor
 
-__all__ = ["HighThroughputExecutor"]
+__all__ = ["HighThroughputExecutor", "ProcessPoolExecutor", "GCExecutor"]
```

Comment:
`GCExecutor` is almost certainly going to conflict with our rename of `FuncXExecutor`.  Yes, it's an entirely different namespace (`funcx_endpoint.*`), but still perhaps unnecessarily confusing for us devs while in the throes of it, and perhaps slightly odd for administrators to think about in terms of `config.py`, yeah?

Perhaps this is a good time to rename this set of work to `*Engine*`?  Something like:
```
funcx_endpoint/
    engines/
        __init__.py
        globuscompute.py
        highthroughput.py
        processpool.py
```

With an `__all__` in `__init__.py`:
```python
from .globuscompute import GlobusComputeEngine
from .highthroughput import HighThroughputEngine
from .processpool import ProcessPoolEngine

__all__ = (GlobusComputeEngine, HighThroughputEngine, ProcessPoolEngine)
```

Reply:
We have consensus over on slack #dev that this renaming makes sense. That said, I think I'd like to get this in and have this reorg done in a follow-up PR. Is that okay with you?

Reply:
As we merged PR #1046 yesterday, I think we can resolve this conversation, yeah?  Feel free to unresolve, if appropriate, and we can discuss further ...